server/index.js

```diff
@@ -12,17 +12,83 @@ app.use(express.static(__dirname + '/../client/dist'));
 app.use(express.json());
 app.use(cors());
 
-
-app.get('/images', (req, res) => {
- console.log("request receiver") 
-	db.find()
+//get all images
+app.get('/images', (req, res) => { 
+	getImages()
   .then(images => {
-    res.json({
-  images
+    res.status(200).json({images})
+  })
+  .catch(err => {
+    res.status(200).send('unable to get all images');
+  })
+})
+
+//get only one image by Id
+app.get('/images/:imageId', (req,res) => {
+  const id = req.params.imageId;
+  getImageById(id)
+    .then(image => {
+      res.status(200).send(image);
+    })
+    .catch(err => {
+      res.status(500).send(`unable to get item with id: ${id} from database`)
+    })
+})
+
+//post a image
+app.post('/images', (req,res) => {
+  const image = new imageModal(req.body);
+  postImage(image)
+    .then(data => {
+      res.status(200).send('item saved to database!')
+    })
+    .catch(err => {
+      res.status(500).send('unable to save to database!')
     })
+})
+
+//update item by imageId
+app.put('/images/:imageId', (req,res) => {
+  //id to find entry to update
+  const id = req.params.imageId;
+  //request body to update the information, and replace if there is a new one needed
+  const image = new imageModal(req.body);
+  updateImage(id,image)
+    .then(data => {res.status(200).send(`item with id: ${id} updated in database`)})
+    .catch(err => {res.status(500).send(`item with id: ${id} NOT updated in database`)})
+})
+
+//partial update with patch request
+app.patch('/images/:imageId', (req,res) => {
+  const id = req.params.imageId;
+  const image = newImageModal(req.body);
+  partialUpdate(id,image)
+    .then(data => {res.status(200).send(`item with id: ${id}`)})
+    .catch(err => {res.status(500).send(`item with id ${id}`)})
+})
+
+//delete an item by imageId
+app.delete('/images/:imageId', (req,res) => {
+  const id = req.params.productId;
+  deleteImage(_id: id)
+  .then(data => {
+    res.status(200).status.send('item deleted from database');
+  })
+  .catch(err => {
+    res.status(500).send('unable to delete item from database')
   })
 })
 
+//delete all images
+app.delete('/images', (req,res) => {
+  deleteAllImages()
+    .then(data => {
+      res.status(200).send('deleted all images in collection');
+    })
+    .catch(err => {
+      res.status(500).send('unable to delete all images from the collection');
+    })
+})
 
```

Comment:
To reiterate, no .then or .catch blocks will run because the database is not sending back anything!

Reply:
ok cool thanks, do you have any ideas how I can send the status code back as a response? or who I will be able to get the then and catch blocks to run when there is no data being sent back from the database.